http/src/request/guild/create_guild.rs

```diff
@@ -182,17 +339,42 @@ impl<'a> CreateGuild<'a> {
 
     /// Set the roles to create with the guild.
     ///
-    /// The maximum number of roles that can be provided is 250.
+    /// The maximum number of roles that can be provided is 250. A `Vec` of [`RoleFragment`]
+    /// structs must be provided. These will be turned into instances of `Role`s that are sent to
+    /// Discord. If specifying `Some` for [`RoleFragment.id`], do not use the id of 1.
     ///
     /// # Errors
     ///
     /// Returns [`CreateGuildError::TooManyRoles`] if the number of roles is
     /// over 250.
     ///
     /// [`CreateGuildError::TooManyRoles`]: enum.CreateGuildError.html#variant.TooManyRoles
-    pub fn roles(mut self, roles: Vec<Role>) -> Result<Self, CreateGuildError> {
-        if roles.len() > 250 {
-            return Err(CreateGuildError::TooManyRoles { roles });
+    /// [`RoleFragment`]: struct.RoleFragment.html
+    /// [`RoleFragment.id`]: struct.RoleFragment.html#structfield.id
+    pub fn roles(mut self, fragments: Vec<RoleFragment>) -> Result<Self, CreateGuildError> {
+        if fragments.len() > 250 {
+            return Err(CreateGuildError::TooManyRoles { roles: fragments });
+        }
+
+        let mut roles: Vec<Role> = fragments
+            .iter()
+            .map(|f| Role {
+                color: f.color.unwrap_or(0),
+                hoist: f.hoist,
+                id: f.id.unwrap_or_else(|| RoleId(2)),
+                managed: false,
+                mentionable: f.mentionable,
+                name: f.name.clone().unwrap_or_else(String::new),
+                permissions: f.permissions.unwrap_or_else(Permissions::empty),
+                position: f.position.unwrap_or(0),
+            })
+            .collect();
+
+        // if there are already overrides, retain the first role in the vec.
+        // discord understands the first role to be overrides for @everyone.
+        if self.fields.roles.is_some() {
```

Comment:
if let

Reply:
I run into ownership issues here, which is why there is a call to `take`